src/main/java/de/tum/in/www1/artemis/service/iris/IrisGPT3_5Service.java

```diff
@@ -0,0 +1,200 @@
+     *
+     * @param session The session to generate a response for.
+     * @return A {@link CompletableFuture} that will be completed with the response.
+     */
+    @Override
+    @Async
+    public CompletableFuture<String> getResponse(IrisSession session) {
+        try {
+            Map<String, Object> requestParams = createRequestParams(session.getMessages());
+            ObjectMapper jsonMapper = new ObjectMapper();
+            String requestBody = jsonMapper.writeValueAsString(requestParams);
+            log.debug("Request body: {}", requestBody);
+
+            HttpEntity<String> request = new HttpEntity<>(requestBody, createHeaders());
+            String responseBody = restTemplate.postForObject(apiURL.toString(), request, String.class);
+            log.debug("Response body: {}", responseBody);
+
+            OpenAIChatResponseDTO response = jsonMapper.readValue(responseBody, OpenAIChatResponseDTO.class);
+            String content = response.getChoices().get(0).getMessage().getContent();
+
+            return CompletableFuture.completedFuture(content);
+        }
+        catch (JsonProcessingException e) {
+            log.error(e.getMessage(), e);
+            return CompletableFuture.failedFuture(e);
+        }
+    }
+
+    /**
+     * Gets a {@link Map} of request parameters to be sent to GPT-3.5. These parameters will be converted into a
+     * JSON object and sent in the request body.
+     *
+     * @param conversationHistory The messages that have been sent in the conversation so far
+     * @return The API request parameters
+     */
+    private Map<String, Object> createRequestParams(List<IrisMessage> conversationHistory) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("model", model);
+        parameters.put("messages", convertMessages(conversationHistory));
+        parameters.put("temperature", temperature);
+        parameters.put("max_tokens", maxGeneratedTokens);
+        if (stopSequences.length > 0)
+            parameters.put("stop", stopSequences);
+        return parameters;
+    }
+
+    /**
+     * Creates a HttpHeaders object with the Content-Type set to application/json and the Authorization header set to
+     * the API key, which varies depending on the model used.
+     *
+     * @return A HttpHeaders object to be used in a request to a LLM API
+     */
+    private HttpHeaders createHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("api-key", apiKey);
+        return headers;
+    }
+
+    /**
+     * Converts the system instructions and conversation history into GPT3_5's message format.
+     *
+     * @param conversationHistory The conversation history to convert.
+     * @return The converted messages.
+     */
+    private static List<Map<String, String>> convertMessages(List<IrisMessage> conversationHistory) {
```

Comment:
Hey Michael, some method names weren't clear for me when I first read the code. I mean you described everything really good in the comment above it, but when the method is used in other places, it's maybe not obvious what it does. Maybe you could change the method name, so it's clear to what format it converts the messages e.g., convertMessagesToGPT3_5Format or something like that. The method that you call in this function (toRoleMessage) also converts Messages to this GPT3_5 format right? So maybe you could also change that method name, to something like toGPT3_5MessageFormat or something instead of toRoleMessage?